### PR TITLE
Print out predicted flakes from input data

### PR DESCRIPTION
**Pull request by stefwalter**
This changes around so that the validation set and predict
set are the same input data. In addition we print out information
about what in the predict set is a flake.

### CODE REVIEW

**Comment by stefwalter** [comment]
This can be tested with: https://stefw.fedorapeople.org/2017-07-28-open-cockpit-test-data.jsonl.gz

File: cp_test_nn/predict.py [hunk 1]
```diff
@@ -1,23 +1,39 @@
...
-            successes += 1
-        elif pred == [1]:
-            false_positives += 1
-        else:
-            false_negatives += 1
-    print('Success rate: ', successes/total_count)
-    print('False positives: ', false_positives/total_count)
-    print('False negatives: ', false_negatives/total_count)
+        sys.stdout.write("{0} {1} {2} {3} {4}\n".format(
+            pred and "FLAKE  " or "UNKNOWN",
```
Comment (bkabrda) [inline cp_test_nn/predict.py]: So `pred` is a list of predictions - since this is a prediction for one item, it will be a list containing exactly one prediction. It will always be 0 or 1, since we're only predicting membership of two classes - "flake" (`pred == [1]`) and "not flake" (`pred == [0]`).
I think what we really want here now is to do `pred = nn.predict([item])[0]` on the line above and then you should be able to do this (+ also replace "UNKNOWN" for "NOT FLAKE" or something similar).

Does this make sense?

**Comment by bkabrda** [comment]
@stefwalter this PR LGTM except one small inline comment. Also, in the meanwhile I committed a patch that implements logging, so you'll need to rebase. Sorry for that :/

File: cp_test_nn/predict.py (hunk 1 above)
Comment (stefwalter) [reply cp_test_nn/predict.py]: Yes, that makes sense. Sorry for misunderstanding.

As an aside, I'm also playing around with predict_proba here ... I'd like to print out some probability number about how sure we are that this is a flake.  Or does that even make sense with MLPClassifier?

Comment (bkabrda) [reply cp_test_nn/predict.py]: No problem :)

The neural network actually always gives a probability as a prediction, the scikit-learn predict method just rounds it to 0 or 1. So it definitely makes sense to look at probabilities. So we can definitely do something like "prob < 0.2 => not flake; prob > 0.8 => flake; 0.2 <= prob <= 0.8 => unknown". I haven't yet looked at the actual probabilities, so I'm not sure what would be reasonable margins. I can definitely look into this.

**Comment by bkabrda** [comment]
LGTM, merging. Thanks!
